blog/15-02-2023-how-to-tokenize-physical-assets-on-celo/how-to-tokenize-physical-assets-on-celo.md

```diff
@@ -0,0 +1,300 @@
+	return "ipfs://sdsdsdfdfdfdfdfcx2323sds";
+}
+
+function pause() public onlyOwner {
+	_pause();
+}
+
+function unpause() public onlyOwner {
+	_unpause();
+}
+
+function safeMint(address to, string memory uri) public onlyOwner {
+	uint256 tokenId = _tokenIdCounter.current();
+	_tokenIdCounter.increment();
+	_safeMint(to, tokenId);
+	_setTokenURI(tokenId, uri);
+}
+
+function _beforeTokenTransfer(address from, address to, uint256 tokenId, uint256 batchSize) internal whenNotPaused override{
+	super._beforeTokenTransfer(from, to, tokenId, batchSize);
+}
+
+
+// The following functions are overrides required by Solidity.
+function _burn(uint256 tokenId) internal override(ERC721, ERC721URIStorage) {
+	super._burn(tokenId);
+}
+
+function tokenURI(uint256 tokenId) public view override(ERC721, ERC721URIStorage) returns (string memory) {
+	return super.tokenURI(tokenId);
+}
+
+}
+
+ ```
+ 
+
+ Let's go over each item in the code to help you if you are a new Solidity developer:
+
+  *-   // SPDX-License-Identifier: MIT:* This is a comment that specifies the license under which the code is released.
+  
+*-   pragma solidity ^0.8.9;:* This line specifies the version of Solidity being used.
+    
+*-   import "@openzeppelin/contracts/token/ERC721/ERC721.sol";:* This is an import statement that brings in the ERC721 contract from the OpenZeppelin library. ERC721 is the standard for NFTs on Ethereum.
+    
+*-   import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";:* This is another import statement that brings in the ERC721URIStorage contract from OpenZeppelin. This extension allows for the storing and retrieving of metadata associated with the token.
+    
+*-   import "@openzeppelin/contracts/security/Pausable.sol";:* This import statement brings in the Pausable contract from OpenZeppelin. This allows the contract owner to pause and unpause token transfers in case of emergencies.
+    
+*-   import "@openzeppelin/contracts/access/Ownable.sol";:* This import statement brings in the Ownable contract from OpenZeppelin. This contract implements a basic access control mechanism that restricts access to certain functions to the contract owner.
+    
+*-   import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Burnable.sol";:* This is another import statement that brings in the ERC721Burnable contract from OpenZeppelin. This extension adds the ability to burn or destroy tokens.
+    
+*-   import "@openzeppelin/contracts/utils/Counters.sol";:* This import statement brings in the Counters library from OpenZeppelin. Counters is used to keep track of the token IDs.
+    
+
+The contract itself is called TokenizationNFT and inherits from several of the imported contracts, including ERC721, ERC721URIStorage, Pausable, Ownable, and ERC721Burnable.
+
+The using Counters for Counters.Counter; statement is used to create a Counters.Counter variable that will keep track of the token IDs.
```

Comment:
The using Counters for Counters.Counter; statement is used to create a Counters. The counter variable will keep track of the token IDs.